### PR TITLE
Github Actions Artifact Publishing

### PR DESCRIPTION
**Pull request by Michael7371**
Added GitHub action workflow to publish a java artifact to the GitHub repository whenever a release is created.

### CODE REVIEW

**Review by dmccoystephenson — CHANGES_REQUESTED** [review]
This looks good to me; has it been tested?

**Comment by Michael7371** [comment]
> This looks good to me; has it been tested?

Yes, here is a published artifact from a test run: https://github.com/CDOT-CV/jpo-s3-deposit/packages/2175004

**Review by dmccoystephenson — APPROVED** [review]
Looks good!

**Review by drewjj — APPROVED** [review]
Looks good

**Comment by iyourshaw** [comment]
Looks good!  

Only comment is I notice in your test run the version is a SNAPSHOT, which is fine for testing, but I'd recommend removing that from the version number in the pom for the actual release. (SNAPSHOT makes the artifacts mutable which could be confusing to consumers of them)

**Comment by Michael7371** [comment]
> Looks good!
> 
> Only comment is I notice in your test run the version is a SNAPSHOT, which is fine for testing, but I'd recommend removing that from the version number in the pom for the actual release. (SNAPSHOT makes the artifacts mutable which could be confusing to consumers of them)

Good idea! I added a step to the pipeline that should remove the '-SNAPSHOT' from the pom file before building and publishing the artifact.

File: .github/workflows/artifact-publish.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Publish Java Package
+
+on:
+  push:
+    tags:
+      - 'jpo-s3-deposit-*'
```
Comment (payneBrandon) [inline .github/workflows/artifact-publish.yml:6]: Did we want to ensure we only are creating artifact for "release" tags?

**Review by payneBrandon — CHANGES_REQUESTED** [review]
Just the one question, looks good!

File: .github/workflows/artifact-publish.yml (hunk 1 above)
Comment (Michael7371) [reply .github/workflows/artifact-publish.yml:6]: I figured that we should go with the current convention in the USDOT repository. All of their release tags are prefixed with the name of the repository. Do we plan to create separate release tags in the USDOT branches?

**Review by payneBrandon — APPROVED** [review]
lgtm!

**Pull request merged**